pkg/da/data_availability_header.go

```diff
@@ -124,11 +119,13 @@ func DataAvailabilityHeaderFromProto(dahp *daproto.DataAvailabilityHeader) (dah
 	dah.RowsRoots = dahp.RowRoots
 	dah.ColumnRoots = dahp.ColumnRoots
 
-	return dah, dah.ValidateBasic()
+	return dah, dah.ValidateBasic(minSquareSize, maxSquareSize)
 }
 
 // ValidateBasic runs stateless checks on the DataAvailabilityHeader.
-func (dah *DataAvailabilityHeader) ValidateBasic() error {
+func (dah *DataAvailabilityHeader) ValidateBasic(minSquareSize, maxSquareSize int) error {
+	maxExtendedSquareWidth := minSquareSize * 2
+	minExtendedSquareWidth := maxSquareSize * 2
```

Comment:
[blocking] there is a subtle bug here.
```suggestion
	minExtendedSquareWidth := minSquareSize * 2
	maxExtendedSquareWidth := maxSquareSize * 2
```

Reply:
wow, good catch!!! This would have broken celestia-node for sure. We don't call validate basic iirc, since the nodes are the ones generating the DAH.

We should think about how this will get used there as well, and making this function state-full might not be the best approach. I think we can stick to using the default max and mins. If we want to change the max and mins beyond those values in governance, then we can force a hardfork. wdyt?